### ISSUE TITLE
Remove labels if we don't need them on an issue anymore (so remove failed if it's now passing, etc...)

### ISSUE THREAD

**Issue by DarthHater** [sonatype-nexus-community/the-cla#50]
Basically right now, labels are getting created and added, but not removed when a state SWITCHES. This should be likely very easy, but I've eaten my words before.

So if someone has a failed PR, with a "This has failed" (not exact text) label on it, and it now no longer fails, we add the "This is good" label, but we should remove the "This has failed" label if it's still applied to the issue. Vice versa. 

Basically, you don't end up in this state (unless you are a jerk and apply them yourself):

<img width="329" alt="Screen Shot 2022-01-21 at 11 09 51 AM" src="https://user-images.githubusercontent.com/5544326/150593510-79c8b891-fa4f-47fe-a6d4-ff699e153a31.png">

**Comment by bhamail** [comment]
Should be fixed by 60057ae0
